### PR TITLE
end to end tests for APPS-1004 and APPS-1005

### PR DESCRIPTION
**Pull request by rodicasutu**
End to end tests for APPS-1004 and APPS-1005

### CODE REVIEW

File: amps/ags/rm-automation/rm-automation-community-rest-api/src/test/java/org/alfresco/rest/rm/community/recordcategories/MoveRecCategoriesWithRSTests.java [hunk 1]
```diff
@@ -0,0 +1,287 @@
...
+/**
+ * Tests for moving record categories between record categories with different retention schedule
+ */
+public class MoveRecCategoriesWithRSTests extends BaseRMRestTest
+{
+    /**
+     * list with the root categories to be deleted after running the test
+     */
+    private List<String> categoriesToBeDeleted = new ArrayList<>();
+    ;
```
Comment (cagache) [inline amps/ags/rm-automation/rm-automation-community-rest-api/src/test/java/org/alfresco/rest/rm/community/recordcategories/MoveRecCategoriesWithRSTests.java:64]: Useless ;

File: amps/ags/rm-automation/rm-automation-community-rest-api/src/test/java/org/alfresco/rest/rm/community/recordcategories/MoveRecCategoriesWithRSTests.java [hunk 2]
```diff
@@ -0,0 +1,287 @@
...
+        assertTrue(nonElRecord.getProperties().getRecordSearchDispositionEvents().contains(RMEvents.OBSOLETE.getEventName()),
+                "Event list doesn't contain the event from the current RS ");
+    }
+
+    @AfterClass (alwaysRun = true)
+    public void cleanupMoveRecCategoriesWithRSTests()
+    {
+        categoriesToBeDeleted.forEach(cat -> getRestAPIFactory().getRecordCategoryAPI().deleteRecordCategory(cat));
+    }
+}
```
Comment (cagache) [inline amps/ags/rm-automation/rm-automation-community-rest-api/src/test/java/org/alfresco/rest/rm/community/recordcategories/MoveRecCategoriesWithRSTests.java:287]: I would try to somehow remove some of the duplicated code. For example I would create both root categories in before method and delete them in after method and I would create a helper method for creating the  subcategory-record folder-records structure.

**Review by cagache — APPROVED** [review]

**Review by shubhamjain10 — APPROVED** [review]
Similar observation as Claudia regarding the preconditions for test. Rest all the changes seems completely fine to me.

File: amps/ags/rm-automation/rm-automation-community-rest-api/src/test/java/org/alfresco/rest/rm/community/recordcategories/MoveRecCategoriesWithRSTests.java (hunk 2 above)
Comment (rodicasutu) [reply amps/ags/rm-automation/rm-automation-community-rest-api/src/test/java/org/alfresco/rest/rm/community/recordcategories/MoveRecCategoriesWithRSTests.java:268]: Thanks for the suggestions! I have update the tests

**Comment by rodicasutu** [comment]
> Similar observation as Claudia regarding the preconditions for test. Rest all the changes seems completely fine to me.

Updates added

**Pull request merged**